src/components/SimulatorArea.tsx

```diff
@@ -33,18 +38,21 @@ const Canvas = styled('canvas', {
   touchAction: 'none',
 });
 
-export class SimulatorArea extends React.Component<SimulatorAreaProps> {
+export class SimulatorArea extends React.Component<SimulatorAreaProps, SimulatorAreaState> {
   private containerRef_: HTMLDivElement;
   private canvasRef_: HTMLCanvasElement;
 
   constructor(props: SimulatorAreaProps) {
     super(props);
-    this.state = {};
+    this.state = { loading: true };
   }
 
   componentDidMount() {
     // TODO: If simulator initialization fails, we should show the user an error
     Sim.Space.getInstance().ensureInitialized()
+      .then(() => {
+        this.setState({ loading: false });
+      })
       .catch(e => console.error('Simulator initialization failed', e));
```

Comment:
In case initialization fails, it'd be nice to show a general error message, so that the loader doesn't spin forever

Reply:
Good call - I'll add a timeout with a message

Reply:
Should be resolved now, although we may want to change the error message